### ISSUE TITLE
include Samsung's aggressive battery optimization

### ISSUE THREAD

**Issue by Rudxain** [Rudxain/blog#1]
For S-rants

https://github.com/urbandroid-team/dont-kill-my-app/issues/307#issuecomment-827649020

Also add some commentary about it

**Comment by Rich968** [comment]
Many of these problems are probably just 'bugs' in the software/hardware on the phones. e.g very short notifications (several characters long) are delayed or queued on Samsung A10 but fine on Samsung A04e. Also echo cancelling is not working on A10 but works well on A04e. Foreground service (when playing YouTube stuff) , gets killed and immediately restarted on A10 but again fine on A04e!   Incidently on a very cheap Chinese smartphone - around £40 all this works fine! What is going on? Any comments from Samsung?...........

**Comment by Rudxain** [comment]
> probably just 'bugs'

I agree. Samsung can't be so ill-intentioned. As _Hanlon's Razor_ states: "Don't attribute to malice what can be explained by incompetence"

> What is going on? Any comments from Samsung?...........

I guess the higher-ups just want a lot of software done, even if it's low-quality

**Comment by Rich968** [comment]
Time-scale pressure during product development, combined with inadequate testing also probably responsible. Yes, variability in the competence of engineering teams, and associated management across products as well.  A04e  very much better than 'flaky' A10. I'm finding that their A8 Tablet is so-far OK.....

**Comment by buawf** [comment]
> Many of these problems are probably just 'bugs' in the software/hardware on the phones. e.g very short notifications (several characters long) are delayed or queued on Samsung A10 but fine on Samsung A04e. Also echo cancelling is not working on A10 but works well on A04e. Foreground service (when playing YouTube stuff) , gets killed and immediately restarted on A10 but again fine on A04e! Incidently on a very cheap Chinese smartphone - around £40 all this works fine! What is going on? Any comments from Samsung?...........

 
  
  Are both devices One Ui  Core  ?

**Comment by Rich968** [comment]
Samsung A10 device is One UI 3.1 Android version 11. Samsung A04e device is One UI 5.1 Android version 13. Incidentally accessing echo cancelling completely differently, (code change) now works on all phones. Notification delays (irespective of message length) seem to be very variable on the A10 and something is definitely not right! All the problems seem to be limited to the A10 device. Don't know whether this is an Android 11 problem or a hardware problem with the phone,  All other devices that I have tried, work fine with my Apps. The A10 seems to be the only device with issues. Foreground killing on A10 is unrelated to App content! Even an 'empty' foreground service that only displays a notification gets killed! Incidently every other foreground service app that I have tried, suffers from same problem, both with the notifications and with the App killing, so not my code!

**Comment by Rich968** [comment]
Samsung A04e now upgraded to Android 14. Delayed notifications are back! However fixed by enabling vibrate. Incidently doing this also on the Samsung A10 Android 11, has also fixed delayed notification problem.  Is this a deliberate feature? Seems like notifications are in a queue of length two before they appear, unless vibrate is enabled. Very odd! Problem only occurs when the phone is asleep.

**Comment by Rudxain** [comment]
I want to thank everyone here for their info and suggestions! If anyone is willing to open a PR for S-rants, I'd appreciate it.

Honestly, I have no idea how to properly incorporate the contents of this thread into the page, so I might just "add a link and call it a day", but simply linking here seems too lazy.

Thoughts? 👀🤔

**Comment by Rudxain** [comment]
[Found another power-saving "feature"](https://eu.community.samsung.com/t5/galaxy-note-series/games-lag-unless-my-finger-is-pressed-to-screen/td-p/2011231)

Can confirm this also happens on A31

**Comment by Rudxain** [comment]
This repo will be "deleted" (merged) soon:
- Rudxain/Rudxain.github.io#17

I'm considering to transfer this issue (again 😅)

Just a heads-up to anyone subscribed, **links will break** (more details in linked issue)
